### DISCUSSION TITLE
How to change the soil depth in e3sm-2.1.0?

### DISCUSSION THREAD

**Discussion by LianGong98** [E3SM-Project/E3SM#5954] [E3SM model help]
Hello,
  I am try to modify the soil depth in E3SM, in the default model setting, there are a total of 10 soil layers 3.5m? if I want to change it to 6 m, how should I do ? 
  Extremely grateful for any assistance

**Comment by rljacob** [comment]
@bishtgautam any advice on this?

**Comment by bishtgautam** [comment]
@LianGong98, The z-value at the centroid of the soil layer (`zsoi`) is computed at [components/elm/src/main/initVerticalMod.F90#L133-L135](https://github.com/E3SM-Project/E3SM/blob/master/components/elm/src/main/initVerticalMod.F90#L133-L135) and you can modify that code block accordingly.

**Comment by mpaiao** [comment]
@bishtgautam I was trying some tests with ELM-FATES to address #7297, and updated the code above to the following:

```fortran
  subroutine initVertical(bounds, snow_depth, thick_wall, thick_roof)
    [...]
    real(r8)              :: scalez = 0.030_r8 ! Soil layer thickness discretization (m)
    real(r8)              :: scalee = 0.56_r8  ! Soil layer exponential factor
    [...]
       do j = 1, nlevgrnd
          zsoi(j) = scalez*(exp(scalee*(j-0.5_r8))-1._r8)    !node depths
       enddo
    [...]
  end subroutine initVertical
```

So zsoi for layer j=10 (nlevsoi) would be 6.101516 m, and much deeper than the default. The layers in the output look as expected with these updated parameters. However, when I looked at variables like `WATSAT`, instead of making the active soil zone deeper, it reduced the number of soil layers considered active to 8 (zsoi(8) = 1.97059 m), which is the closest zsoi to the original zsoi(10) (2.864607113 m). Are there other places I should update if I want to run a quick test with a deeper active soil zone? Thanks!
